File: doc-examples/example-jackson-cbor-java/src/main/java/example/Book.java
package example;

import io.micronaut.serde.annotation.Serdeable;

@Serdeable // <1>
public class Book {
    private final String title;
    private final int quantity;

    public Book(String title, int quantity) {
        this.title = title;
        this.quantity = quantity;
    }

    public String getTitle() {
        return title;
    }

    public int getQuantity() {
        return quantity;
    }
}

File: doc-examples/example-bson-python/src/main/python/example/Book.py
from dataclasses import dataclass
from typing import Annotated

from micronaut.serde.annotation import Serdeable
from org.bson.codecs.pojo.annotations import BsonProperty


@Serdeable  # <1>
@dataclass
class Book:
    title: str
    quantity: Annotated[int, BsonProperty("qty")]  # <2>
